tests/unit/type/stream-cgroups.tcl

```diff
@@ -212,7 +212,7 @@ start_server {
         set rd [redis_deferring_client]
         $rd XREADGROUP GROUP mygroup Alice BLOCK 0 STREAMS mystream ">"
         r DEL mystream
-        assert_error "*no longer exists*" {$rd read}
```

Comment:
did we change an error code in any of these?
i.e. did the error code in all 6 cases modified in this file remain as it was (either `NOGROUP` or `WRONGTYPE`)?
i think that if we did change the error code, we must find a way to fix it.

but even if it's just the text, maybe it could be a little bit confusing.
i.e. before, it was clear that the error was from the unblocking process...
on the other hand, the command could have just as well arrive from the original command (depending on the order of execution).
@guybe7 @itamarhaber WDYT?

Reply:
@oranagra I think the error code was changed in case of streamgroup blocking operation which was unblocked due to stream deletion (we used to return -UNBLOCKED) I can try and fix that (maybe by looking at the UNBLOCKED flag in the stream process command) in order to prevent this from being breaking change.

Reply:
@oranagra / @madolson / @guybe7   - 
Basically I can align the errors of a deleted/change type stream (either by multi/exec, flushall, swapdb or stream delete)
Currently the implementation is to issue 
``
UNBLOCKED the stream key no longer exists
``
in the following cases:
1. the stream key has been deleted (ie. calling DEL)
2. the stream and group existed but the key type was changed by overriding it (ie. with set command)
3. The key not longer exists after we swapdb with a db which does not contains this key
4. after swapdb when the new db has this key but with different type.

Now I would only like to ask at this point what do you find a better option:
1. we align to the current behavior (always error with -UNBLOCKED)
while this makes sense that a different error is used (otherwise why this command blocked and not exit with error in first place) this will prevent the user from understanding exactly what happened to the stream key... was it overridden with different type or was it deleted...) - I can keep the error type and use different error massages in this case, but I wonder if applications will match error massages.
one thing to note is that it is basically non-deterministic error reporting since (for example) a blocking command might have been racing with a key deletion/override 
2. We can keep my current implementation that will not issue -UNBLOCKED but will use the same error type as today
which will be more useful for a user to distinguish the 2 cases, but is less "ideal" in terms of the rational to "why did this blocked on the first case"
     
     wanted to catch your thoughts on that.

Reply:
i prefer (2)

this idea of this PR is to re-process blocked clients, which means it can fail in exactly the same way as the non-blocking variant. 
when a user executes a blocking command it has to expect and handle errors that may arise in case the command didn't block in the first place, so i don't see this as a breaking change

Reply:
I also favor the 2nd option as it is more informative, and am similarly of the opinion that error type changes do not fall into the breaking catergory.

Reply:
although part of me wants to resist interface changes due to refactoring, i tend to agree that we should take the second option, for the same reason described about (race).
i hope we're not missing anything, and some odd reason why a user should distinguish between these two cases (being blocked and released, or not blocked in the first place).

@ranshid please make sure this is clearly described in the top comment.

Reply:
@oranagra I extended the previous comment about that in the top comment.